### ISSUE TITLE
Colors for Categories

### ISSUE THREAD

**Issue by TJScalzo** [tholman/inspiring-online#337]
I played around for a bit and I came up with two main ideas for adding color to categories:
![Screen Shot 2020-01-26 at 1 19 04 PM](https://user-images.githubusercontent.com/15825214/73139709-d5f56280-403e-11ea-8352-b5ecdfccda3f.png)![Screen Shot 2020-01-26 at 1 19 18 PM](https://user-images.githubusercontent.com/15825214/73139710-d5f56280-403e-11ea-86dc-07ef6246d654.png) I made these by drawing a colored triangle over the pattern and setting the global blend mode to multiply so the pattern shows through.

Personally, I prefer the gradient since it has some bright, highly saturated color without being overwhelming. However, on mobile it has some problems because the bright part overlaps with the colorful category title which ends up needing an outline to be readable.

Of course, these are just two basic ideas that can be modified or combined if necessary. It's incredible the flexibility you have when using HTML canvas!

If these ideas seem reasonable and/or match the aesthetic of the site, I can clean up my code and push it to a branch for perusing.

**Comment by TJScalzo** [comment]
On my fork I've made a [colors branch](https://github.com/TJScalzo/inspiring-online/tree/colors) where I've pushed the code that created the images above. I also made it redraw the triangles differently when the window is made small enough. This is done to avoid the category title overlapping too much with the gradient. The commit messages have some more details.

**Comment by TJScalzo** [comment]
It still looks pretty nice when a triangle has color without a pattern. The main thing to do would be picking out colors that are unique and look nice together.

For example, here's Art with a random blue:
![Screen Shot 2020-02-02 at 6 19 40 PM](https://user-images.githubusercontent.com/15825214/73617059-d797cb00-45e8-11ea-86da-76adf559b231.png)
![Screen Shot 2020-02-02 at 6 28 57 PM](https://user-images.githubusercontent.com/15825214/73617203-1c703180-45ea-11ea-9101-ab287a1d6daf.png)
@tholman I guess I'm just looking for some direction. There are a lot of options that can be explored, I just need to know which one you think is best for the site.

**Comment by tholman** [comment]
Oooh, these are looking good! I like the gradients (and the category being colored)

I wonder if something like this https://generativeartistry.com/tutorials/triangular-mesh/ would work... where the tiles vary in different shades of the color, but tilt towards lightness at the end of them. That could add a little more dynamicy to the page... if that makes any sense!

**Comment by TJScalzo** [comment]
> Oooh, these are looking good! I like the gradients (and the category being colored)

I'm glad you like it!


> I wonder if something like this https://generativeartistry.com/tutorials/triangular-mesh/ would work... where the tiles vary in different shades of the color, but tilt towards lightness at the end of them. That could add a little more dynamicy to the page... if that makes any sense!

I'm not sure I understand what you mean. Would the random triangles replace the patterns that we already have or are they just for categories with no patterns? I'll try out some options. I'm not super familiar with HTML Canvas and generative design so this will be quite a learning experience for me.

**Comment by tholman** [comment]
Yeah, I think they would replace the patterns that are already there. Might be easier to maintain even less images, and have all that come from the code. 

Happy to help with any canvas stuff if you want to give it an initial attempt! Always fun to learn, so I'm here to help!

**Comment by TJScalzo** [comment]
Alright, so based on the tutorial you linked, I made these:
White Lines ⬇️![Screen Shot 2020-02-04 at 11 28 31 PM](https://user-images.githubusercontent.com/15825214/73810982-28085780-47a6-11ea-9f75-ed054d836143.png)
Color Lines ⬇️![Screen Shot 2020-02-05 at 12 23 05 AM](https://user-images.githubusercontent.com/15825214/73813369-c350fb00-47ad-11ea-9460-54665ec53f51.png)

It took a couple of tries and failed messes, but I've finally got it working! The way I'm doing it feels a little hacky, but I like how it turned out. I've made the mobile version in pretty much the same way as before, with the color of the triangles becoming much more muted. Let me know what you think!

**Comment by tholman** [comment]
These. Look. Amazing! I'm very into the first one... we should definitely run with that!

Happy to review any code and do any fixups or guide you through it. 

Also wondering what it would be like if we made the edges the nodes of the triangles, so they wouldn't be perfectly flat... I guess that would involve building the initial grid of nodes in a slightly different way. Something we could explore afterwards anways

**Comment by TJScalzo** [comment]
> Happy to review any code and do any fixups or guide you through it.

I’m glad to hear it. I’m not super familiar with optimizing code so there might be a more efficient way to do this. 

> Also wondering what it would be like if we made the edges the nodes of the triangles, so they wouldn't be perfectly flat... I guess that would involve building the initial grid of nodes in a slightly different way. Something we could explore afterwards anways

That would probably make the edges look nicer since there wouldn’t be just small corners of triangles clipped at the edges. But yeah, we would definitely have to change how the nodes are drawn to do this. Right now I’m just drawing a full grid and then painting over everything outside the main triangle. 

**Question:** Should categories that don’t have a color be flat grey, should they get triangles on top, or should they be a grey gradient? Right now they’re just the same flat grey with no triangles. (If we add colors for each category, there won’t be any categories that need to default to grey. But a fallback is always nice to have.)

**Comment by TJScalzo** [comment]
I've just found a problem with the method I'm using:
![Screen Shot 2020-02-06 at 2 22 15 PM](https://user-images.githubusercontent.com/15825214/73971382-c3a3e000-48ec-11ea-87c6-16992564043b.png)
When on a larger display, the canvii become so large that they overlap with each other vertically. This means the white mask I'm drawing to clear the extra triangles/gradient also clears the tip of the triangle above it. I'm not sure if there's a solution since posts on Stack Overflow say there's no way to erase things from a canvas other than using `clearRect()`.

**Comment by TJScalzo** [comment]
I have one last question before I push my code @tholman. What do you want the backup design for categories that don't have a color to be?
Flat Grey: ![Screen Shot 2020-02-06 at 2 54 53 PM](https://user-images.githubusercontent.com/15825214/73973750-2d25ed80-48f1-11ea-8db0-08b5a9bad199.png)
Grey Triangles: ![Screen Shot 2020-02-06 at 2 54 16 PM](https://user-images.githubusercontent.com/15825214/73973774-37e08280-48f1-11ea-97a6-384422698402.png)

Personally, I like the triangles because they make everything feel more consistent. Let me know!

**Comment by tholman** [comment]
Ooh good solve with the desitnation-in. If I try to do the non masking by building the triangle grid differently, it would also solve that problem... but we can see when we come into it.

Ya, the triangles for in grey for sure are best!

**Comment by TJScalzo** [comment]
I just pushed my code for making the triangles. Also, I added a bunch of colors:
![Screen Shot 2020-02-06 at 8 34 36 PM](https://user-images.githubusercontent.com/15825214/73993594-a557d780-4920-11ea-9ae2-93bb100903f2.png)
The nice thing about the triangles is that two different colors never butt up against each other so they don't have to all match. I kind of just picked colors that I thought looked nice with the title of each category. If you have any strong feelings about any of them, I'm happy to change them.
